### ISSUE TITLE
YunoHost should ask for the preferred language/locale during cli install

### ISSUE THREAD

**Issue by fcnjd** [YunoHost/issues#1903]
### Describe the bug
Normally, debian-based systems allow to change the default system language by running the command "dpkg-reconfigure locales" with root privileges. However, after YunoHost is installed, this isn't possible anymore, because even if the command is ran and is set to change the default language, it has no effect. After executing "locale" the environment variables still aren't changed, therefore I have the impression that YunoHost overrides them.

### Context

- Hardware: VPS bought online
- YunoHost version: 4.2.8.3
- I have access to my server: Through SSH
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: no

### Steps to reproduce

    ```sh
    sudo dpkg-reconfigure locales
    ```

### Expected behavior
The system should set the environment variables for the selected default language propperly, as with normal debian-based systems

**Comment by alexAubin** [comment]
Can you elaborate how you confirm that the changes are not properly taken into account ?

**Comment by alexAubin** [comment]
(bumpitybump)

**Comment by fcnjd** [comment]
Oh, it looks like my comment by email reply wasn't transfered properly, I'll quote it again.

> Sure: I've reconfigured the locales package, selected de_DE.UTF-8 as the default locale to use, and then did a server reboot and logged in again. And here's the output of the locale command:
root@fcnjd:~# locale
LANG=de_DE.UTF-8
LANGUAGE=
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=en_US.UTF-8
root@fcnjd:~#
In contrast, I've done exactly the same with a freshly installed server, but built from the same Debian-10 image. There, after the reboot, all variables of the "locale" command are set to de_DE.UTF-8, therefore it must be YunoHost that keeps them from beeing changed properly.

**Comment by alexAubin** [comment]
zblerg i don't know why that would happen ... i can't think of anything funky we would be doing with locales ... I guess maybe there's something in your `.bashrc` or other bash profile files (on your server), and it also depends on how exactly you connect with SSH (your configuration may be forwarding locale vars from your client). Also you can check what's the system default locale with `localectl`, that may provide more clue ..

**Comment by fcnjd** [comment]
Hi,
after a bit of debugging and testing, I found where the bug's coming from: the fix_locales function in the install script. I'll explain why:
On a fresh debian installation, without YunoHost, the file /etc/environment exists, but with size 0, there's no content. Reconfiguring the locales and changing the default language works propperly, but still doesn't modify /etc/environment in any way, no matter what you do in the configuration. However, when you run the YunoHost installer, it modifies the file, and sets en_US.UTF-8 as LC_ALL, in any case. The line of code that does that is
    # If no /etc/environment exists, default to en_US.UTF-8
    [ "$(grep LC_ALL /etc/environment)" ] || echo 'LC_ALL="en_US.UTF-8"' >> /etc/environment
So, and now you can set whatever locale you want as your default, the LC_ALL entry, made by the YunoHost installer overrides it. You can actually fix this by just making the environment file empty again, than the changes are read again as they should get normally. Anyway, I beleive this should be fixed somehow, that the environment file isn't touched in such a way, because a user should be able to change the language afterwards, without having it always overwritten.

I didn't make a PR with the line removed from the script, as I'm not sure why exactly you write the variable into the environment file (although I'd be interested why). Anyway, I'd be happy if you could fix that.

**Comment by alexAubin** [comment]
This dates back from a while ago : https://github.com/YunoHost/install_script/commit/21ab5a714f2dfb6969e7ceefacef2f5a75b39102

Basically locales are hell and it's an absolute mess to understand what's the correct way to configure them (manual configuration ain't obvious, but then comes trying to automatize stuff on setups which you don't 100% know, i.e. VPS providers may have preconfigured stuff, some other wont), and in particular perl will display a shitload of warnings if locales happen to be misconfigured.

At some point we got mad and added that trick, which ain't optimal considering that it indeeds hardcode en_US as the locale ... but less worse than having no locale at all ...

tl;dr : it all boils down to the fact that we should properly ask admins what are their prefered languages during install (or use the locale from the current shell), but hmpf ... anyway, this issue probably affects only power-users considering that the webadmin language is based on the browser's language
